src/utils/assign.js

```diff
@@ -0,0 +1,11 @@
+// Object.assign() is not available in IE11. And the babel compiled output for object spread
+// syntax checks a bunch of Symbol stuff and is almost a kb. So this function is the smaller replacement.
+export function assign() {
+  for (let i = arguments.length - 1; i > 0; i--) {
+    for (let key in arguments[i]) {
```

Comment:
isn't this vulnerable to prototype pollution?

do we need to add a check here for `if key === "__proto__" continue;` ?

Reply:
I don't understand prototype pollution, could you teach me what it is? Have only seen it in npm audit warnings and github security warnings.

Reply:
the main idea is that if you blindly map over all the keys in an object, someone can pass an object that looks like the following:

`{__proto__.toString: () => alert('bad')`

now, because we didn't prevent that from happening in our library, all JS objects running in the same environment will have isAdmin = true because we set the object's prototype to that value. E.g. 

```js
const myObject = {}
console.log(myObject) //alerts 'bad'
```

or something like that. :D

I'm not a pro in this area, though, so I may be misrepresenting the issue.

Reply:
Updated